plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -73,8 +73,8 @@ protected final String getRefreshedLocation(boolean invalidateCaches)
         }
         Table table = getTable();
 
-        if (isPrestoView(table) && isHiveOrPrestoView(table)) {
-            // this is a Hive view, hence not a table
+        if (isPrestoView(table) && isHiveOrPrestoViewOrPrestoMaterializedView(table)) {
```

Comment:
I originally did what you suggested and tests failed, see: https://github.com/trinodb/trino/actions/runs/4833558229/jobs/8614428829

```
Error:  io.trino.faulttolerant.iceberg.TestIcebergParquetFaultTolerantExecutionConnectorTest.testMaterializedViewGracePeriod  Time elapsed: 2 s  <<< FAILURE!
...
Caused by: io.trino.plugin.iceberg.UnknownTableTypeException: Not an Iceberg table: tpch.test_mv_grace_period_dz8ejt94jq
	at io.trino.plugin.iceberg.catalog.hms.AbstractMetastoreTableOperations.getRefreshedLocation(AbstractMetastoreTableOperations.java:81)
	at io.trino.plugin.iceberg.catalog.AbstractIcebergTableOperations.refresh(AbstractIcebergTableOperations.java:134)
	at io.trino.plugin.iceberg.catalog.AbstractIcebergTableOperations.current(AbstractIcebergTableOperations.java:117)
	at io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog.lambda$loadTable$11(TrinoHiveCatalog.java:373)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog.loadTable(TrinoHiveCatalog.java:371)
	at io.trino.plugin.iceberg.IcebergMetadata.getTableHandle(IcebergMetadata.java:369)
```

It is because `IcebergMetadata#getTableHandle` catches `TableNotFoundException` and returns null but throws `UnknownTableTypeException`. 

I can do:
```
        if (!isIcebergTable(table)) {
            throw new TableNotFoundException(getSchemaTableName());
        }
```

Reply:
ok, so let's keep `if isViewOrMaterializedView` here